src/webgpu/api/validation/encoding/render_bundle.spec.ts

```diff
@@ -51,3 +60,245 @@ g.test('render_bundles,device_mismatch')
 
     encoder.validateFinish(!mismatched);
   });
+
+g.test('color_formats_mismatch')
+  .desc(
+    `
+    Tests executeBundles cannot be called with render bundles that do match the colorFormats of the
+    render pass. This includes:
+    - formats don't match
+    - formats match but are in a different order
+    - formats match but there is a different count
+    `
+  )
+  .params(u =>
+    u.combineWithParams([
+      {
+        bundleFormats: ['bgra8unorm', 'rg8unorm'] as const,
+        passFormats: ['bgra8unorm', 'rg8unorm'] as const,
+        _compatible: true,
+      }, // control case
+      {
+        bundleFormats: ['bgra8unorm', 'rg8unorm'] as const,
+        passFormats: ['bgra8unorm', 'bgra8unorm'] as const,
+        _compatible: false,
+      },
+      {
+        bundleFormats: ['bgra8unorm', 'rg8unorm'] as const,
+        passFormats: ['rg8unorm', 'bgra8unorm'] as const,
+        _compatible: false,
+      },
+      {
+        bundleFormats: ['bgra8unorm', 'rg8unorm', 'rgba8unorm'] as const,
+        passFormats: ['rg8unorm', 'bgra8unorm'] as const,
+        _compatible: false,
+      },
+      {
+        bundleFormats: ['bgra8unorm', 'rg8unorm'] as const,
+        passFormats: ['rg8unorm', 'bgra8unorm', 'rgba8unorm'] as const,
+        _compatible: false,
+      },
+    ])
+  )
+  .fn(async t => {
+    const { bundleFormats, passFormats, _compatible } = t.params;
+
+    const bundleEncoder = t.device.createRenderBundleEncoder({
+      colorFormats: bundleFormats,
+    });
+    const bundle = bundleEncoder.finish();
+
+    const encoder = t.createEncoder('render pass', {
+      attachmentInfo: {
+        colorFormats: passFormats,
+      },
+    });
+    encoder.encoder.executeBundles([bundle]);
+
+    encoder.validateFinish(_compatible);
+  });
+
+g.test('depth_stencil_formats_mismatch')
+  .desc(
+    `
+    Tests executeBundles cannot be called with render bundles that do match the depthStencil of the
+    render pass. This includes:
+    - formats don't match
+    - formats have matching depth or stencil aspects, but other aspects are missing
+    `
+  )
+  .params(u =>
+    u.combineWithParams([
+      { bundleFormat: 'depth24plus' as const, passFormat: 'depth24plus' as const }, // control case
+      { bundleFormat: 'depth24plus' as const, passFormat: 'depth16unorm' as const },
+      { bundleFormat: 'depth24plus' as const, passFormat: 'depth24plus-stencil8' as const },
+      { bundleFormat: 'stencil8' as const, passFormat: 'depth24plus-stencil8' as const },
+    ])
```

Comment:
could be wrong, but I *think* you can put `as const` all the way at the end, after the array brackets.
```suggestion
      { bundleFormat: 'depth24plus', passFormat: 'depth24plus' }, // control case
      { bundleFormat: 'depth24plus', passFormat: 'depth16unorm' },
      { bundleFormat: 'depth24plus', passFormat: 'depth24plus-stencil8' },
      { bundleFormat: 'stencil8', passFormat: 'depth24plus-stencil8' },
    ] as const)
```